### PR TITLE
Document MIDI zero velocity quirk more completely

### LINKED ISSUE #76678: 4.0, MIDI input: releasing key on MIDI device returns event.message 0x9 (MIDI_MESSAGE_NOTE_ON); should be 0x8 (MIDI_MESSAGE_NOTE_OFF)
### Godot version

4.0.2.stable.mono

### System information

Windows 10

### Issue description

Hello! I'm playing around with the MIDI input in Godot 4, and I've run into the issue where releasing a key on my MIDI keyboard does not return event.message 0x8, MIDI_MESSAGE_NOTE_OFF, but rather 0x9, which is MIDI_MESSAGE_NOTE_ON.

This is not the case in Godot 3. I created an identical script in each to test this out, and releasing the key in 3.x always returns 0x8, whereas releasing the key in 4 always returns 0x9.

### Steps to reproduce

```
func _ready():
    OS.open_midi_inputs()

func _input(event):
    if event is InputEventMIDI:
        print(event.message)
```

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by voidshine**
*Bugsquad edit:*
- Closes #76678

### CODE REVIEW

**Comment by voidshine** [comment]
@akien-mga recommended expanding the documentation, and this PR does just that.

https://github.com/godotengine/godot/issues/76678#issuecomment-1547689054

File: doc/classes/InputEventMIDI.xml [hunk 1]
```diff
@@ -81,7 +81,8 @@
 			Returns a value indicating the type of message for this MIDI signal. This is a member of the [enum MIDIMessage] enum.
 			For MIDI messages between 0x80 and 0xEF, only the left half of the bits are returned as this value, as the other part is the channel (ex: 0x94 becomes 0x9). For MIDI messages from 0xF0 to 0xFF, the value is returned as-is.
 			Notes will return [constant MIDI_MESSAGE_NOTE_ON] when activated, but they might not always return [constant MIDI_MESSAGE_NOTE_OFF] when deactivated, therefore your code should treat the input as stopped if some period of time has passed.
-			For more information, see the MIDI message status byte list chart linked above.
+			Some MIDI devices may send [constant MIDI_MESSAGE_NOTE_ON] with zero velocity instead of [constant MIDI_MESSAGE_NOTE_OFF].
+			For more information, see 'velocity' note below and the MIDI message status byte list chart linked above.
```
Comment (akien-mga) [inline doc/classes/InputEventMIDI.xml:85]:
```suggestion
			For more information, see the note in [member velocity] and the MIDI message status byte list chart linked above.
```

File: doc/classes/InputEventMIDI.xml [hunk 2]
```diff
@@ -90,7 +91,7 @@
...
 		</member>
 		<member name="velocity" type="int" setter="set_velocity" getter="get_velocity" default="0">
-			The velocity of the MIDI signal. This value ranges from 0 to 127. For a piano, this corresponds to how quickly the key was pressed, and is rarely above about 110 in practice. Note that some MIDI devices may send a [constant MIDI_MESSAGE_NOTE_ON] message with zero velocity and expect this to be treated the same as a [constant MIDI_MESSAGE_NOTE_OFF] message, but device implementations vary so Godot reports event data exactly as received.
+			The velocity of the MIDI signal. This value ranges from 0 to 127. For a piano, this corresponds to how quickly the key was pressed, and is rarely above about 110 in practice. Note that some MIDI devices may send a [constant MIDI_MESSAGE_NOTE_ON] message with zero velocity and expect this to be treated the same as a [constant MIDI_MESSAGE_NOTE_OFF] message, but device implementations vary so Godot reports event data exactly as received. Depending on the hardware and the needs of the game/app, this MIDI quirk can be handled robustly with a couple lines of script (check for [constant MIDI_MESSAGE_NOTE_ON] with velocity zero).
```
Comment (akien-mga) [inline doc/classes/InputEventMIDI.xml:94]: I'd suggest splitting the note about this quirk for readability:.
```suggestion
			The velocity of the MIDI signal. This value ranges from 0 to 127. For a piano, this corresponds to how quickly the key was pressed, and is rarely above about 110 in practice.
			[b]Note:[/b] Some MIDI devices may send a [constant MIDI_MESSAGE_NOTE_ON] message with zero velocity and expect this to be treated the same as a [constant MIDI_MESSAGE_NOTE_OFF] message, but device implementations vary so Godot reports event data exactly as received. Depending on the hardware and the needs of the game/app, this MIDI quirk can be handled robustly with a couple lines of script (check for [constant MIDI_MESSAGE_NOTE_ON] with velocity zero).
```

**Review by akien-mga — APPROVED** [review]

**Comment by voidshine** [comment]
Thanks for the improvements! Both suggestions are committed.

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by voidshine** [comment]
Squashed & force pushed, thanks for the reminder. :)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs another rebase to fix an unrelated CI issue (temporary mismatch between this repo and the godot-cpp one used in a test at the time you last rebased).

**Comment by voidshine** [comment]
Okay, synced and rebased.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks and congrats on your first merged Godot PR! 🎉

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**